R/summary.R

```diff
@@ -21,6 +21,7 @@
 #' sumcorp <- summary(corp) # (quietly) assign the results
 #' sumcorp$Types / sumcorp$Tokens # crude type-token ratio
 summary.corpus <- function(object, n = 100, tolower = FALSE, showmeta = TRUE, ...) {
+    .Deprecated("textstat_summary()")
```

Comment:
There is no rush, but transition takes long time. Our transition strategy depends on how the simplified `summary()` will look like.

Reply:
Agreed. I was thinking of something like this for the `summary()` methods. A bit like `print()` but instead of showing a glimpse of the contents, we would show a "summary" of the core object and its metadata. Something like:

```r
> summary(data_corpus_inaugural)
Corpus consisting of 58 documents, four docvars.
Document length: mean 2,565, IQR 1,568, 3,131 tokens.
Title: "US presidential inaugural address speeches"
Created 2020-02-27.
 
> tokens(data_corpus_inaugural, remove_punct = TRUE) %>%
     summary()
Tokens consisting of 58 documents, four docvars.
Document length: mean 2,333, IQR 1,433, 2,982 tokens.
Title: "US presidential inaugural address speeches"
Created 2020-07-25.
```
etc

Reply:
It is similar to` meta(data_corpus_inaugural)`. How about something like this?
```
> summary(attr(data_corpus_inaugural, "docvars")[1:3])
   docname_                     docid_       segid_ 
 Length:58          1789-Washington: 1   Min.   :1  
 Class :character   1793-Washington: 1   1st Qu.:1  
 Mode  :character   1797-Adams     : 1   Median :1  
                    1801-Jefferson : 1   Mean   :1  
                    1805-Jefferson : 1   3rd Qu.:1  
                    1809-Madison   : 1   Max.   :1  
                    (Other)        :52  

> summary(attr(corpus_reshape(data_corpus_inaugural), "docvars")[1:3])
   docname_                   docid_         segid_      
 Length:5018        1841-Harrison: 210   Min.   :  1.00  
 Class :character   1925-Coolidge: 196   1st Qu.: 22.00  
 Mode  :character   1909-Taft    : 158   Median : 48.50  
                    1929-Hoover  : 158   Mean   : 56.36  
                    1889-Harrison: 157   3rd Qu.: 83.00  
                    1845-Polk    : 153   Max.   :210.00  
                    (Other)      :3986 
```

Reply:
Something like that could work. Let's close this PR and move the discussion to a new issue on changing and extending `summary()` and differentiating it from the new `textstat_summary()`.